### PR TITLE
Handle error with user hint, if feedback endpoint is not available.

### PR DESCRIPTION
**Pull request by lulo4907**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/modules/feedback/components/FeedbackBoxContainer.js [hunk 1]
```diff
@@ -72,7 +73,8 @@ export class FeedbackBoxContainer extends React.Component<PropsType, StateType>
       if (postFeedbackDataOverride) {
         postFeedbackDataOverride(feedbackData)
       } else {
-        await createFeedbackEndpoint(cmsApiBaseUrl).request(feedbackData)
+        const res = await createFeedbackEndpoint(cmsApiBaseUrl).request(feedbackData)
+        if (res) { return res }
```
Comment (klinzo) [inline src/modules/feedback/components/FeedbackBoxContainer.js:77]: No need to check for the value of the result of the request, it always returns `{}` for the feedback endpoint if it does not fail. Just return `true` if you don't have to catch an error and `false` if you have to. You should also handle `postFeedbackDataOverride` above.

File: src/modules/feedback/components/FeedbackBox.js [hunk 2]
```diff
@@ -73,9 +75,10 @@ export class FeedbackBox extends React.PureComponent<PropsType> {
           commentMessage={isPositiveRatingSelected ? t('positiveComment') : t('negativeComment')}
           onCommentChanged={onCommentChanged}
           required={!isPositiveRatingSelected} />
+        <Description>{requestError ? t('failedSendingFeedback') : null}</Description>
         <TextButton
           disabled={!isPositiveRatingSelected && !comment}
-          onClick={onSubmit}
+          onClick={requestError ? null : onSubmit}
```
Comment (klinzo) [inline src/modules/feedback/components/FeedbackBox.js:81]: This does not seem right (especially in terms of ui/ux, but also where this is handled). You shouldn't just disable a button without giving the user a hint that it is disabled and providing an option to enable it again. Imo the logic to show that sending the feedback failed should not reside here, but instead either in the refactored `FeedbackThanksMessage` (possibly with a try again button) or a entire new component, both rendered in the `FeedbackModal`. If you use the suggestion of the enum, you could switch between its value to render either a `FeedbackBox`, a `FeedbackThanksMessage` or a `FeedbackError` there.

File: src/modules/feedback/components/FeedbackModal.js [hunk 3]
```diff
@@ -64,7 +64,9 @@ type StateType = {|
 export class FeedbackModal extends React.Component<PropsType, StateType> {
   state = { feedbackSent: false }
 
-  handleSubmit = () => this.setState({ feedbackSent: true })
+  handleSubmit = (arg: any) => {
```
[the remark this answers is not in the document]
Comment (lulo4907) [reply src/modules/feedback/components/FeedbackModal.js]: changed

File: src/modules/feedback/components/FeedbackBoxContainer.js (hunk 1 above)
Comment (lulo4907) [reply src/modules/feedback/components/FeedbackBoxContainer.js]: changed

File: src/modules/feedback/components/FeedbackBox.js (hunk 2 above)
Comment (lulo4907) [reply src/modules/feedback/components/FeedbackBox.js]: changed

**Pull request closed without merging**